### ISSUE TITLE
[Bug] Registration function with automatic `launch_op` doesn't work for multi-device tensors correctly

### ISSUE THREAD

**Issue by omilyutin-tt** [tenstorrent/tt-metal#15442]
Operation registration with automatic `launch_op` makes a couple of assumptions that might not be true for multi-device tensors:
1. The tensor returned from an operation resides on a single device. During the registration, we create a wrapper for "multi device" tensor, distribute the work to workers, each worker then creates a "multi device" tensor from the operation itself, and finally we try to link those individual tensors to the top level multi device tensor, which leads to an error.
2. The number workers attached to an input tensor will equal to the number of workers participating in the operation. This isn't the case for operations like `{full,empty,zeros,ones}_like`, as we might want to create a multi-device tensor *like* the single-device input tensor.

We don't need to use the automatic `launch_op` for `{full,empty,zeros,ones}_like`, as they already distribute the allocation / data transfers work asynchronously to the worker threads. However, other use cases may require this fixed.

**Comment by prajaramanTT** [comment]
@omilyutin-tt is this still an open issue ? If not, can you please close this ? Thanks

**Comment by omilyutin-tt** [comment]
Yes, this is still an issue, not high priority as it doesn't block anything AFAIK.
